src/to-have-value.js

```diff
@@ -23,7 +17,7 @@ export function toHaveValue(htmlElement, expectedValue) {
   const expectsValue = expectedValue !== undefined
   return {
     pass: expectsValue
-      ? isEqualWith(receivedValue, expectedValue, compareArraysAsSet)
```

Comment:
This change would mean we can no longer match arrays regardless of order, as per my [comment](https://github.com/testing-library/jest-dom/pull/166#issuecomment-555398045). But we wouldn't be able to support jest matchers without it, e.g. `expect.arrayContaining(['second', 'third'])`